chain/indexer/src/streamer/mod.rs

```diff
@@ -163,8 +165,6 @@ pub async fn build_streamer_message(
                 .push(IndexerExecutionOutcomeWithReceipt { execution_outcome, receipt });
         }
 
-        chunk_receipts.extend(chunk_non_local_receipts);
-
         // Find the shard index for the chunk by shard_id
         let shard_index = protocol_config_view
```

Comment:
This is likely a bug (unrelated to your PR). The header.shard_id may contain shard id from the previous shard layout in case of a resharding with a missing chunk in the first block. Do you have some free cycles to look into it? If not I can give it a stab. The correct way would be to enumerate chunks to get the shard index this way. It's terrible, I know.

Reply:
chunk is guaranteed to be "new chunk" in this case, so shard_id should be from the new shard layout, so this should work